### PR TITLE
Pytest added for function process_workflow_list

### PR DESCRIPTION
**Pull request by egenc**
Purpose of this PR is to add tests for the function`Cloudos.process_workflow_list` in CloudOS. 

Test data is obtained from `Cloudos.process_workflow_list` function and values are masked as `XXX`.
Test data:
- tests/test_data/process_workflow_list_initial_request.json
- tests/test_data/process_workflow_list_results.csv
- tests/test_data/process_workflow_list_results_FULL.csv

Tests are made for checking how the function `process_workflow_list` operating:
- `test_process_workflow_list` to check `all_fields=False` (specific columns)
- `test_process_workflow_list_all_fields` to check `all_fields=True` (all columns)

To run the test:
```
python3 -m pytest -s -v
```
Expected output:
image

Minor update:
- `.gitignore` is updated for adding Mac file metadata by appending this file:
```
# .DS_Store
.DS_Store
._.DS_Store
**/.DS_Store
**/._.DS_Store
```

### CODE REVIEW

File: tests/test_clos/test_process_workflow_list.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import requests, json
+from cloudos.clos import Cloudos
+import pytest
+import requests_mock
+
```
Comment (danielboloc) [inline tests/test_clos/test_process_workflow_list.py:5]:
```suggestion

```

File: tests/test_clos/test_process_workflow_list.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import pandas as pd
+
+INPUT_JSON = "tests/test_data/process_workflow_list_initial_request.json"
+output_df = pd.read_csv("tests/test_data/process_workflow_list_results.csv")
+output_df_full = pd.read_csv("tests/test_data/process_workflow_list_results_FULL.csv")
+
+
```
Comment (danielboloc) [inline tests/test_clos/test_process_workflow_list.py:12]:
```suggestion

```

File: tests/test_clos/test_process_workflow_list.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    test_workspace_id = 1
+    with open(INPUT_JSON) as json_data:
+        d = json.load(json_data)
+    with requests_mock.Mocker() as mock:
+        mock.get(f"http://test_cloud_os/api/v1/jobs?teamId={test_workspace_id}",
+                 json=d)
+        r = requests.get(f"http://test_cloud_os/api/v1/jobs?teamId={test_workspace_id}")
+    return r
+
+def test_process_workflow_list(mocked_requests_get):
```
Comment (danielboloc) [inline tests/test_clos/test_process_workflow_list.py:24]:
```suggestion
def test_process_workflow_list_all_fields_false(mocked_requests_get):
    """Test function parameter 'all_fields=False'"""
```

File: tests/test_clos/test_process_workflow_list.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    df = Cloudos.process_workflow_list(mocked_requests_get, all_fields=False)
+    assert df.equals(output_df)
+
+def test_process_workflow_list_all_fields(mocked_requests_get):
```
Comment (danielboloc) [inline tests/test_clos/test_process_workflow_list.py:28]:
```suggestion
def test_process_workflow_list_all_fields_true(mocked_requests_get):
    """Test function parameter 'all_fields=True'"""
```

**Review by danielboloc — COMMENTED** [review]
@egenc looks good, I have left some tiny suggestions regarding doc strings and function names

**Review by danielboloc — APPROVED** [review]

File: tests/test_clos/test_process_workflow_list.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import requests, json
```
Comment (dapineyro) [inline tests/test_clos/test_process_workflow_list.py:1]:
```suggestion
import requests
import json
```
Please, use one import per line

File: tests/test_clos/test_process_workflow_list.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+from cloudos.clos import Cloudos
+import pytest
+import requests_mock
+import pandas as pd
+
+INPUT_JSON = "tests/test_data/process_workflow_list_initial_request.json"
+output_df = pd.read_csv("tests/test_data/process_workflow_list_results.csv")
+output_df_full = pd.read_csv("tests/test_data/process_workflow_list_results_FULL.csv")
+
```
Comment (dapineyro) [inline tests/test_clos/test_process_workflow_list.py:10]:
```suggestion


```
Two blank lines

File: tests/test_clos/test_process_workflow_list.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+def mocked_requests_get():
+    test_workspace_id = 1
+    with open(INPUT_JSON) as json_data:
+        d = json.load(json_data)
+    with requests_mock.Mocker() as mock:
+        mock.get(f"http://test_cloud_os/api/v1/jobs?teamId={test_workspace_id}",
+                 json=d)
+        r = requests.get(f"http://test_cloud_os/api/v1/jobs?teamId={test_workspace_id}")
+    return r
+
```
Comment (dapineyro) [inline tests/test_clos/test_process_workflow_list.py:21]:
```suggestion


```
Two blank lines

File: tests/test_clos/test_process_workflow_list.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+def test_process_workflow_list_all_fields_false(mocked_requests_get):
+    """Test function parameter 'all_fields=False'"""
+    df = Cloudos.process_workflow_list(mocked_requests_get, all_fields=False)
+    assert df.equals(output_df)
+
```
Comment (dapineyro) [inline tests/test_clos/test_process_workflow_list.py:26]:
```suggestion


```
Two blank lines

**Review by dapineyro — CHANGES_REQUESTED** [review]
Thanks @egenc ! The code is great, just some minor style suggestions. Regarding the data you added in `test_data`, could it be possible to trim it to just contain one entry (one workflow)?

In addition, I observed that the structure of `process_workflow_list_initial_request.json` is not identical to the real JSON: you have changed the nested levels by using `.` notation. It does not affect the current function, as it's doing an equivalent `pandas.json_normalize()` but the contents of the function may change in the future, so it would be better to have a more realistic input.

File: tests/test_clos/test_process_workflow_list.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+def test_process_workflow_list_all_fields_false(mocked_requests_get):
+    """Test function parameter 'all_fields=False'"""
+    df = Cloudos.process_workflow_list(mocked_requests_get, all_fields=False)
+    assert (df.columns == output_df.columns).any()
+
+
+def test_process_workflow_list_all_fields_true(mocked_requests_get):
+    """Test function parameter 'all_fields=True'"""
+
```
Comment (dapineyro) [inline tests/test_clos/test_process_workflow_list.py:32]:
```suggestion
```

**Review by dapineyro — CHANGES_REQUESTED** [review]
This one looks great @egenc . Just check the unresolved conversation and also the new suggestion

File: tests/test_clos/test_process_workflow_list.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
+"""pytest for fucntion process_workflow_list"""
```
Comment (dapineyro) [inline tests/test_clos/test_process_workflow_list.py:1]:
```suggestion
"""pytest for method Cloudos.process_workflow_list"""
```
Although it's not really necessary to add this docstring to this simple file.

File: tests/test_clos/test_process_workflow_list.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    """Test function parameter 'all_fields=False'"""
+    df_ = Cloudos.process_workflow_list(mocked_requests_get, all_fields=False)
+    assert (df_.columns == output_df.columns).any()
+
+
+def test_process_workflow_list_all_fields_true(mocked_requests_get):
+    """Test function parameter 'all_fields=True'"""
+    df_ = Cloudos.process_workflow_list(mocked_requests_get, all_fields=True)
+    assert (df_.columns == output_df_full.columns).any()
+
```
Comment (dapineyro) [inline tests/test_clos/test_process_workflow_list.py:37]:
```suggestion
```

**Review by dapineyro — CHANGES_REQUESTED** [review]
Thanks @egenc . Only two very minor suggestions to check :+1:

**Review by dapineyro — APPROVED** [review]
LGTM! :+1:

**Pull request merged**